### DISCUSSION TITLE
Using a labelled self-hosted runner from act

### DISCUSSION THREAD

**Discussion by m-mcgowan** [nektos/act#1907] [General]
I have a job that runs on a self-hosted runner which I've registered in Github.  Is it possible for `act` to  direct the job execution to that runner so I can test the job locally?

Some more details.  I've installed the github self-hosted runner service to a RasPi, which has some specific hardware attached that is used during CI testing. On my development machine, I'd like to be able to have the job run on that self-hosted runner in the same way Github does. 

The workflow file contains:

```
jobs:
   myjob:
          runs-on: [self-hosted, linux, custom_tag]
```

I've seen the -P option that can be used to map these tags to docker images, but I'm not sure how that can help me run the job in a specific runner. 

I'm probably misunderstanding something, so thanks for any help!

**Comment by goberlan** [comment]
It would be so nice if we can get an answer for this. Or some amount of help. Did you ever figure it out @m-mcgowan?

**Reply by iraizo** [reply]
also interested in this feature

**Reply by critical-mass** [reply]
I'm also exploring this option but running into some errors after following this guid: 
https://juliusgamanyi.com/2022/04/27/run-github-actions-workflows-locally-using-act/

**Comment by ChristopherHX** [comment]
> [ self-hosted, linux, custom_tag ]

You can only use a single label of them (I think it doesn't matter which of them you choose) e.g.  `-P self-hosted=<docker image ref>`. If this "bug" is ever going to be fixed then you might need to specify all labels joined by `,`.

I know this is broken for years and does not allow to use different docker images for different multi labels.

**Comment by kristofferarl** [comment]
Also interested in this. Though I have self-hosted runners already hosted in our data-center I would like to use instead of a local docker image.
